### PR TITLE
Release 6.2.1

### LINKED ISSUE #731: SwiftGen 6.2.1
Allright, with the homebrew swift fix PR up, we should be able to release this version soon:
- [x] Homebrew swift fix #730 
- [x] Small code changes and show warnings in xcode fix #729 
- [x] Array fix for inline templates #687 
- [x] JSONSerialization PR #702
- [x] Plist documents PR #732
- [x] Small doc/template tweaks #739 
- [x] Lazy metadata #709 

After those PRs are done, some general housekeeping:
- [x] Check docs (again)
- [x] Check playground code
- [x] Tiny migration guide for json/plist context `documents` being deprecated in favour of `document`
- [x] Update podspec
- [x] Brew path changes/extra files (new json fixtures)

### PR DESCRIPTION
**Pull request by djbe**
Fixes #731.

I noticed that we forgot to add an entry to the changelog for 6.2, so I wrote something quick. Once we're done reviewing the language of it, we should modify the actual release on GitHub as well.

~Creating this as a draft for now, until we're ready to actually release. Note that I'm merging into develop for now, just for the easier diffs.~

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should make sure that <code>Podfile.lock</code> and <code>Package.resolved</code> are changed in sync</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This is a Release PR</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

### ChangeLog check

✅  All entries seems OK (end with period + 2 spaces, correct links)


### Release version check

| Check                                         | Status       |
| --------------------------------------------- | ------------ |
| Bundler installed                             | ✅           |
| SwiftGen.podspec                              | 👉  6.2.1    |
| SwiftGenKit.podspec                           | 👉  6.2.1    |
| SwiftGen & SwiftGenKit versions equal         | ✅           |
| SwiftGen-Info.plist version matches           | ✅           |
| SwiftGenKit-Info.plist version matches        | ✅           |
| StencilSwiftKit up-to-date (latest: 2.7.2)    | ✅           |
| CHANGELOG: Release entry added                | ✅           |
| CHANGELOG: No develop entry                   | ✅           |
| README: pod version ~> 6.0                    | ✅           |


Once you fix those tiny nitpickings above, we should be good to go! 🙌

ℹ️ _I will update this comment as you add new commits_

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,6 +37,10 @@ _None_
 
 ## 6.2.0
 
+⚠️ This minor version contains a lot of deprecations that may apply to your configuration and how you use SwiftGen. With the exception of `swift3` templates (which have been removed), everything should still work as before. But be warned that all features marked as deprecated will be removed in SwiftGen 7.0.
+
+Read the [SwiftGen 6.2 Migration Guide](Documentation/MigrationGuide.md#migrating-to-swiftgen-62) for a list of changes you'll need to apply.
+
```
Comment (AliSoftware) [inline CHANGELOG.md:46]: 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -94,13 +92,13 @@ _None_
   [#609](https://github.com/SwiftGen/SwiftGen/issues/609)
   [#593](https://github.com/SwiftGen/SwiftGen/pull/593)
   [#610](https://github.com/SwiftGen/SwiftGen/pull/610)
-* Strings: the built-in templates now accept a parameter `lookupFunction` for customizing the localization function, check the [template documentation](Documentation/templates/string/) for more information.  
+* Strings: the built-in templates now accept a parameter `lookupFunction` for customizing the localization function, check the [template documentation](Documentation/templates/strings/) for more information.
```
Comment (AliSoftware) [inline CHANGELOG.md:98]: good catch!

File: README.md [hunk 3]
```diff
@@ -248,7 +248,7 @@ Each parser command generally accepts the same options and syntax, and they mirr
 
 ## Choosing your template
 
-SwiftGen is based on templates (it uses [Stencil](https://github.com/kylef/Stencil) as its template engine). This means that **you can choose the template that fits the Swift version you're using** — and also the one that best fits your preferences — to **adapt the generated code to your own conventions and Swift version**.
+SwiftGen is based on templates (it uses [Stencil](https://github.com/stencilproject/Stencil) as its template engine). This means that **you can choose the template that fits the Swift version you're using** — and also the one that best fits your preferences — to **adapt the generated code to your own conventions and Swift version**.
```
Comment (AliSoftware) [inline README.md:251]: Nice catch here too 👍

File: .swiftlint.yml [hunk 4]
```diff
@@ -82,7 +82,7 @@ file_header:
   required_pattern: |
                     \/\/(
                     \/\/ (SwiftGen(Kit)?|.* UnitTests)
-                    \/\/ Copyright © 2019 SwiftGen
+                    \/\/ Copyright © 2020 SwiftGen
```
Comment (AliSoftware) [inline .swiftlint.yml:85]: ~We need to update `SwiftGen.xcodeproj/xcshareddata/IDETemplateMacros.plist` accordingly too~

nvm, you already did it, just missed it in the see of similar changes 👍

**Review by AliSoftware — APPROVED** [review]
Let's do it!

btw, should we prepare a 6.2.2 to ship #709 soon after this 6.2.1? Or do we prefer to have a 6.3.0 with less stuff and only PRs that are ready or almost ready to go there, like #709 and maybe some others, and add a 6.4.0 milestone for all the rest that is currently flagged for 6.3.0 but won't be ready that soon?

**Comment by djbe** [comment]
Well that's the reason this is still a draft PR, in case we want to merge #709 and/or #739 first. If we don't merge those now, then yeah, do 6.2.2 right after that with those changes.

I like the idea of pushing 6.3 out with what's "ready", and potentially 6.4 if needed before we do 7.

**Comment by marcelofabri** [comment]
Thanks @djbe! 🚀

**Pull request merged**